Charts 5.1.0 RNs.mdx (#1449)
TykTechnologies/tyk-docs: developer-support/release-notes/helm-chart.mdx

---

## 5.1 Release Notes

### 5.1.0 Release Notes

#### Release Date xx 2026

#### Release Highlights

> Tyk Charts 5.1.0 enhances Kubernetes deployment flexibility with support for priority classes across all components, ensuring API Gateway services maintain scheduling priority during resource pressure. This release also renews the default certificate in Gateway Docker images to prevent March 2026 expiration and completes the migration to Bitnami Legacy registry for continued chart compatibility. Additionally, the release fixes secret reference handling in bootstrap jobs for improved custom secret management.

Reviewer: ```suggestion
Tyk Charts 5.1.0 enhances Kubernetes deployment flexibility with support for priority classes across all components, ensuring API Gateway services maintain scheduling priority during resource pressure. This release also renews the default certificate in Gateway Docker images to prevent March 2026 expiration and completes the migration to Bitnami Legacy registry for continued chart compatibility. Additionally, the release fixes secret reference handling in bootstrap jobs for improved custom secret management.

This version aligns with Tyk LTS release 5.8.12.
```

```bash
helm repo add tyk-helm https://helm.tyk.io/public/helm/charts/
helm repo update
helm upgrade [RELEASE_NAME] tyk-helm/[CHART_NAME]
```

#### Downloads
- [Source code](https://github.com/TykTechnologies/tyk-charts/archive/refs/tags/v5.1.0.tar.gz)
- [ArtifactHub - tyk-stack](https://artifacthub.io/packages/helm/tyk-helm/tyk-stack/5.1.0)
- [ArtifactHub - tyk-control-plane](https://artifacthub.io/packages/helm/tyk-helm/tyk-control-plane/5.1.0)
- [ArtifactHub - tyk-data-plane](https://artifacthub.io/packages/helm/tyk-helm/tyk-data-plane/5.1.0)
- [ArtifactHub - tyk-oss](https://artifacthub.io/packages/helm/tyk-helm/tyk-oss/5.1.0)

#### Changelog
<a id="Changelog-v5.1.0" data-scroll-offset></a>

##### Changed
<AccordionGroup>

<Accordion title='Renewed Default Certificate in Gateway Docker Images'>
Updated the default self-signed certificate shipped with Gateway Docker images to prevent expiration in March 2026. This certificate is only intended for initial bootstrapping and testing purposes when `global.tls.gateway=true` is enabled in Helm deployments and must not be used in production environments
</Accordion>

<Accordion title='Completed Migration to Bitnami Legacy Registry'>

> Updated all remaining Bitnami chart references in the Developer Portal to use the Bitnami Legacy registry following Bitnami's distribution model changes. This completes the migration from the standard Bitnami registry to the `bitnamylegacy` registry for PostgreSQL and Redis chart dependencies used in Helm deployment examples and testing infrastructure.

Reviewer: ```suggestion
Updated all remaining references to use the Bitnami Legacy registry following Bitnami's distribution model changes. This completes the migration from the standard Bitnami registry to the `bitnamilegacy` registry for PostgreSQL and Redis chart dependencies used in Helm deployment examples and testing infrastructure.
```

</Accordion>

</AccordionGroup>
##### Added

<AccordionGroup>

<Accordion title='Enhanced Pod Scheduling Control with Kubernetes Priority Classes Support'>
Added support for configuring Kubernetes `priorityClassName` across all Tyk components in Helm charts to prevent API Gateway services from being preempted during cluster resource pressure. Users can now assign appropriate priority levels to Tyk Gateway, Dashboard, Pump, Bootstrap jobs, Dev Portal, Operator, and MDCB components through their respective chart values. This enhancement ensures that essential Tyk infrastructure maintains higher scheduling priority than default workloads, improving service availability and preventing unexpected downtime when Kubernetes nodes experience resource constraints.
</Accordion>

</AccordionGroup>

##### Fixed

<AccordionGroup>

> <Accordion title='Fixed Secret Reference Handling for Admin Credentials in Bootstrap Job'>

Reviewer: ```suggestion
<Accordion title='Fixed Secret Reference Handling for Admin Credentials in Portal Bootstrap Job'>
```

> Fixed an issue in the Helm chart where the bootstrap job incorrectly handled secret references for admin email and password configuration. The bootstrap job now properly respects custom secret names for both admin email and password, allowing users to successfully deploy with their own secret management solutions.

Reviewer: ```suggestion
Fixed an issue where the Developer Portal bootstrap job incorrectly handled secret references for admin email and password configuration. The bootstrap job now properly respects custom secret names for both admin email and password, allowing users to successfully deploy with their own secret management solutions.
```

- [ArtifactHub - tyk-oss](https://artifacthub.io/packages/helm/tyk-helm/tyk-oss/5.1.0)

#### Changelog
<a id="Changelog-v5.1.0" data-scroll-offset></a>

##### Changed
<AccordionGroup>

<Accordion title='Renewed Default Certificate in Gateway Docker Images'>
Updated the default self-signed certificate shipped with Gateway Docker images to prevent expiration in March 2026. This certificate is only intended for initial bootstrapping and testing purposes when `global.tls.gateway=true` is enabled in Helm deployments and must not be used in production environments
</Accordion>

<Accordion title='Completed Migration to Bitnami Legacy Registry'>
Updated all remaining references to use the Bitnami Legacy registry following Bitnami's distribution model changes. This completes the migration from the standard Bitnami registry to the `bitnamilegacy` registry for PostgreSQL and Redis chart dependencies used in Helm deployment examples and testing infrastructure.
</Accordion>

</AccordionGroup>
##### Added

<AccordionGroup>

<Accordion title='Enhanced Pod Scheduling Control with Kubernetes Priority Classes Support'>

> Added support for configuring Kubernetes `priorityClassName` across all Tyk components in Helm charts to prevent API Gateway services from being preempted during cluster resource pressure. Users can now assign appropriate priority levels to Tyk Gateway, Dashboard, Pump, Bootstrap jobs, Dev Portal, Operator, and MDCB components through their respective chart values. This enhancement ensures that essential Tyk infrastructure maintains higher scheduling priority than default workloads, improving service availability and preventing unexpected downtime when Kubernetes nodes experience resource constraints.

Reviewer 2: ```suggestion
Added support for configuring Kubernetes `priorityClassName` across all Tyk components in Helm charts to prevent API Gateway services from being preempted during cluster resource pressure. Users can now assign appropriate priority levels to Tyk Gateway, Dashboard, Pump, Bootstrap jobs, Dev Portal, Operator, and MDCB components through their respective chart values. 

This enhancement ensures that essential Tyk infrastructure maintains a higher scheduling priority than default workloads, improving service availability and preventing unexpected downtime when Kubernetes nodes experience resource constraints.
```
